Tests/SwiftFusionTests/Inference/GenericFactorTests.swift

```diff
@@ -19,57 +19,87 @@ import XCTest
 import PenguinStructures
 @testable import SwiftFusion
 
-/// A factor that selects a pose prior using an integer input index.
-fileprivate struct DemoFactor<Pose: LieGroup>: GenericFactor {
-  typealias Variables = Tuple2<Int, Pose>
+/// A factor that specifies a prior on a pose.
+fileprivate struct PriorFactor<Pose: LieGroup>: GenericFactor {
+  typealias Variables = Tuple1<Pose>
 
   let edges: Variables.Indices
 
-  let priors: [Pose]
+  let prior: Pose
 
   func error(_ variables: Variables) -> Double {
-    let (priorIndex, position) = (variables.head, variables.tail.head)
-    return priors[priorIndex].localCoordinate(position).squaredNorm
+    let actualPosition = variables.head
+    return prior.localCoordinate(actualPosition).squaredNorm
+  }
+}
+
+/// A factor that switches between different linear motions based on an integer label.
+fileprivate struct SwitchingLinearMotionFactor<Pose: LieGroup>: GenericFactor {
+  typealias Variables = Tuple3<Int, Pose, Pose>
+
+  let edges: Variables.Indices
+
+  let motions: [Pose]
+
+  func error(_ variables: Variables) -> Double {
+    let (motionLabel, start, end) = (variables.head, variables.tail.head, variables.tail.tail.head)
+    let actualMotion = between(start, end)
+    return motions[motionLabel].localCoordinate(actualMotion).squaredNorm
```

Comment:
Interesting that we do squaredNorm. This is a departure from GTSAM, where we simply return the error vector. We should probably discuss this in more detail.

Reply:
This returns `error` instead of `errorVector` because the protocol for factors with error vectors doesn't exist yet in feature/generic_factor_graph. That protocol is implemented in https://github.com/borglab/SwiftFusion/pull/75. After I merge this PR, I'll update https://github.com/borglab/SwiftFusion/pull/75 and you'll be able to see the `errorVector` methods for these factors.